lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -2447,6 +2447,13 @@ void TypeChecker::diagnoseIfDeprecated(SourceRange ReferenceRange,
   }
 
   auto *ReferenceDC = Where.getDeclContext();
+
+  // If the reference declaration has the `IgnoreDeprecationWarnings` attribute, don't diagnose.
+  if (auto *innerMostCtx = ReferenceDC->getInnermostDeclarationDeclContext()) {
+    auto *decl = innerMostCtx->getAsDecl();
+    if (decl != nullptr && decl->getAttrs().hasAttribute<IgnoreDeprecationWarningsAttr>()) return;
+  }
+
```

Comment:
Ah sorry, should have also said that the `getAsDecl` is not needed anymore as `getInnermostDeclarationDeclContext` already returns a `Decl *`, so you can do:

```suggestion
  // If the reference declaration has the `IgnoreDeprecationWarnings` attribute, don't diagnose.
  if (auto *decl = ReferenceDC->getInnermostDeclarationDeclContext()) {
    if (decl->getAttrs().hasAttribute<IgnoreDeprecationWarningsAttr>())
      return;
  }
```

and same for the other one